src/irmin-pack/pack.ml

```diff
@@ -359,6 +330,9 @@ struct
           Lwt.return_unit)
 
     let sync t =
+      let former_generation = generation t in
+      let generation = IO.force_generation t.pack.block in
+      if former_generation <> generation then Lru.clear t.lru;
       Dict.sync t.pack.dict;
```

Comment:
There is an possible optimisation here: `pack` and `dict` both change generations at the same time, so we can prevent the `dict` to read the generation from file, by passing a `force_refill` argument to `Dict.sync`.  I'm not sure if it is a good idea or not, as `dict` has its own IO and has a `clear` function that updates its generation.